engine.py

```diff
@@ -578,5 +591,30 @@ def show_dialog(self, title, bundle, widget_class, *args, **kwargs):
         # lastly, return the instantiated widget
         return widget
 
+    def _get_dialog_parent(self):
+        """
+        Get the QWidget parent for all dialogs created through show_dialog &
+        show_modal.
+        """
 
-    
+        from tank.platform.qt import QtGui
+
+        parent = None
+
+        # attempt to find the houdini main window for parenting. The default
+        # implementation in tk-core uses the activeWindow which can be None 
+        # and can also be an already open toolkit dialog. 
+        app = QtGui.QApplication.instance()
+        for widget in app.topLevelWidgets():
+
+            # try to get a hold of the main window. it seems to be the only
+            # one with windowIconText set. There should be a better way to do
+            # this.
+            if (widget.isWindow() and 
+                not isinstance(widget, QtGui.QDialog) and
+                widget.windowIconText()):
+                parent = widget
+
+        self.log_debug("Parenting dialog to: %s" % (parent,))
```

Comment:
should this rather say "determined parent dialog for UI elements to be %s" or something like that? Since it isn't actually don't the parenting here..